standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1526,6 +1529,10 @@ private void updateWSCommitIdAndCleanUpMetadata(Statement stmt, long txnid, TxnT
     if (txnType == TxnType.MATER_VIEW_REBUILD) {
       queryBatch.add("DELETE FROM \"MATERIALIZATION_REBUILD_LOCKS\" WHERE \"MRL_TXN_ID\" = " + txnid);
     }
+    if (txnType == TxnType.COMPACTION) {
```

Comment:
It's not the end of the world to add the CQ_TXN_ID column, but we can avoid that and keep things more straightforward (i.e. keep compaction stuff out of generic TxnHandler and limit it to CompactionTxnHandler which was made specifically for updating compaction-related tables) by updating CQ_NEXT_TXN_ID in CompactionTxnHandler instead, and calling it straight from Worker, maybe right between commitTxn and markCompacted. It would be so much simpler.

Reply:
I would prefer this method for two reasons:
1. To work correctly, the first point when we can update the CQ_NEXT_TXN_ID, is inside the commitTxn, when we already have the lock for the txnId sequence. Anywhere before that, there can be a txn, that will have a higher txnId than the saved CQ_NEXT_TXN_ID and still not see the compaction as committed -> causing the Cleaner to clean up deltas, that should be kept for that txn. So if we don't do it in commitTxn, we only can do it after that, but that would mean, that the update won't be part of the transaction, and can fail "silently"
2. In a follow up Jira, I need to provide a HMS notification for every committed compaction, for that I would also need CQ_TXN_ID

Reply:
I think, you could make `updateWSCommitIdAndCleanUpMetadata` protected and overwrite it in CompactionTxnHandler, so we won't scatter compaction related stuff across many classes.